src/main/java/com/google/sps/servlets/VerifyRoomServlet.java

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.data.*;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Queue;
+import java.util.HashMap;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.Key;
+
+/** Servlet that takes in the user's room Id and verifies that it
+ * exists, then prints the json version of all the urls in playlist
+*/
+@WebServlet("/verify-room")
+public final class VerifyRoomServlet extends HttpServlet {
+  private String inputtedUserTag = "user-party-link";
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    response.setContentType("application/json");
+		
+    String currentRoomId = request.getParameter("roomId");
+    // TODO: handle if they inputted a key string that does not exist in datastore
+    Key currentRoomKey = getKeyFromString(currentRoomId);
+    Room currentRoom = Room.fromKey(currentRoomKey);
+
+    // If the user sent in a room id not in the datastore, send them a hardcoded youtube video
+    // TODO: Redirect to a specific page telling the client that they inputted the wrong room id
+    if(currentRoom == null){
+      String jsonOfTempUrl = new Gson().toJson("https://www.youtube.com/embed/Bey4XXJAqS8");
+      response.getWriter().println(jsonOfTempUrl);
```

Comment:
Just return an error and then the client can display an error message / redirect to home page.